aiobaro/core.py

```diff
@@ -995,16 +1026,38 @@ async def set_presence(
             status_msg (str, optional): The status message to attach to this state.
 
         * Matrix Spec
+        13.7.2.1   PUT /_matrix/client/r0/presence/{userId}/status
+        Content-Type: application/json
+
+        Rate-limited:	Yes.
+        Requires auth:	Yes.
         """
-        return MatrixResponse(httpx.Response(status_code=404, json={}))
+
+        return await self.auth_client(
+            "PUT",
+            f"presence/{user_id}/status",
+            json=dict(
+                filter(
```

Comment:
bad pattern which avoid debug practice. it's more evidence and clean if you create a variable in this case

Reply:
Thanks for the tip ! We will have to fix this in more places